mock_ros2_project/service_client.py

```
@@ -7,18 +7,25 @@
 """
 
 import sys
+from typing import Optional
+
 import rclpy
 from rclpy.node import Node
 from example_interfaces.srv import AddTwoInts
 
 
 class ServiceClient(Node):
-    """Service client node that calls addition service."""
+    """Service client node that calls addition service.
+    
+    This node creates a service client that sends requests to the
+    AddTwoInts service and handles the responses.
+    """
     
-    def __init__(self):
+    def __init__(self) -> None:
+        """Initialize the service client node."""
         super().__init__('service_client')
         
-        # Create client
+        # Create client for the AddTwoInts service
         self.cli = self.create_client(AddTwoInts, 'add_two_ints')
         
         # Wait for service to be available
```

`wait_for_service()` inside a tight loop can stall your node forever if the service never comes up. Consider adding a maximum retry count or overall timeout so startup fails fast instead of hanging indefinitely.